src/expressions/loop-expr.md

```diff
@@ -118,6 +138,37 @@ for n in 1..11 {
 assert_eq!(sum, 55);
 ```
 
+A for loop is equivalent to the following block expression.
+
+```rust
+'label: for PATTERN in iter_expr {
+    /* loop body */
+}
+```
+
+is equivalent to
+
+```rust
+{
+    let result = match IntoIterator::into_iter(iter_expr) {
+        mut iter => 'label: loop {
+            let next;
```

Comment:
Could we simplify this by `let PAT = match ...`?

Reply:
I can't work out how `let mut next = ...` would be different, but `let PAT = match ` would have a different drop order when `PAT` is something like `(x, _)`.

Reply:
[This](https://play.rust-lang.org/?gist=aa45f97595faf63a6e6a48e6adfefb83&version=stable&mode=debug&edition=2015) is why it can't be simplified.